scipy/stats/_continuous_distns.py

```diff
@@ -6123,6 +6144,21 @@ def _stats(self, b, moments='mv'):
     def _entropy(self, c):
         return 1 + 1.0/c - np.log(c)
 
+    def fit(self, data, *args, **kwds):
+        parameters = _check_fit_input_parameters(self, data, args, kwds)
+        data, fshape, floc, fscale = parameters
+        if floc is None:
+            raise RuntimeError("`floc` must be fixed to obtain a "
+                               "well defined solution.")
+        if np.any(data - floc < (fscale if fscale else 0)):
+            raise FitDataError("pareto", lower=0, upper=np.inf)
+        data = data - floc
+        if fscale is None:
+            fscale = np.min(data)
+        if fshape is None:
+            fshape = 1/((1/len(data)) * np.sum(np.log(data/fscale)))
```

Comment:
Is it worth avoiding the division with `(np.sum(np.log(data)) - np.ndarray.size(data)) * np.log(fscale))` or `np.sum(np.log(data) - np.log(fscale))`

Reply:
With a regular dataset I don't see any difference between the resulting shapes' quality of fit, 

```
In [120]: import numpy as np
     ...: from scipy.stats import pareto
     ...: data = pareto.rvs(size=1000, b=3)
     ...: 
     ...: floc = 0
     ...: fscale = np.min(data)
     ...: 
     ...: 
     ...: f1 = (1/ len(data)) * (np.sum(np.log(data)) - data.size * np.log(fscale))
     ...: f2 = (1/ len(data)) * np.sum(np.log(data) - np.log(fscale))
     ...: f0 = (1/ len(data)) * np.sum(np.log(data/fscale))
     ...: f1 == f2 == f0
     ...: 
     ...: args = [data, (pareto._fitstart(data), )]
     ...: func = pareto._reduce_func(args, {})[1]
     ...: 
     ...: f0_LL = func((f0, floc, fscale), data)
     ...: f1_LL = func((f1, floc, fscale), data)
     ...: f2_LL = func((f1, floc, fscale), data)

In [121]: f0_LL
Out[121]: 1549.4373258810017

In [122]: f1_LL
Out[122]: 1549.4373258810017

In [123]: f2_LL
Out[123]: 1549.4373258810017
```

And in terms of speed I'm not sure that the difference is significant
```
In [128]: %timeit (1/ len(data)) * np.sum(np.log(data/fscale))
13.1 µs ± 401 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)

In [129]: #f1

In [130]: %timeit (1/ len(data)) * (np.sum(np.log(data)) - data.size * np.log(fscale))
12.5 µs ± 402 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)

In [131]: #f2

In [132]: %timeit (1/ len(data)) * np.sum(np.log(data) - np.log(fscale))
13.5 µs ± 263 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)
```


Am I missing another benefit for avoiding the division?

Reply:
I was guessing it might make a small difference to speed to avoid division but you put that theory to rest. Thanks for testing.

Reply:
Of course - I appreciate the suggestion!